packages/handler-graphql/src/responses.ts

```diff
@@ -16,14 +16,29 @@ export class ErrorResponse {
         code: string;
         message: string;
         data?: any;
+        stack?: string;
     };
-    constructor(params: ErrorResponseParams) {
+    constructor(params: ErrorResponseParams | Error | (Error & ErrorResponseParams)) {
         this.data = null;
-        this.error = {
-            code: params.code || defaultParams.code,
-            message: params.message || defaultParams.message,
-            data: params.data || defaultParams.data
-        };
+
+        if (params instanceof Error) {
```

Comment:
I decided to dive a bit deeper into this code block, as it seemed a bit more complex than it should be.

Ended up with the following solution:

```ts
interface ErrorResponseParams {
  message: string;
  code?: string;
  data?: any;
  stack?: string;
}

const defaultParams: ErrorResponseParams = {
  code: "",
  message: "",
  data: null,
  stack: null,
};

export class ErrorResponse {
  data: null;
  error: {
    code: string;
    message: string;
    data: any;
    stack: string;
  };
  constructor(params: ErrorResponseParams) {
    this.data = null;

    const debug = process.env.DEBUG === "true";

    let stack = defaultParams.stack;
    if (debug && params.stack) {
      stack = params.stack;
    }

    this.error = {
      code: params.code || defaultParams.code,
      message: params.message || defaultParams.message,
      data: params.data || defaultParams.data,
      stack
    };
  }
}
```

The most important change I guess would be the `constructor(params: ErrorResponseParams) {` simplification. Ultimately, it doesn't matter if the constructor accepts `Error` or not. It's important that `Error` is just compatible with the `ErrorResponseParams` interface, which is true. Once that's done, the below code gets simplified significantly.


I'm hoping I didn't miss something here, let me know if I did!